_posts/2020-09-15-data-sharing-at-scale.md

```diff
@@ -0,0 +1,307 @@
+
+The source accounts will need:
+
+- _Somewhere to store their data_ – We can store CSV files in an S3 Bucket
+- _A way to catalog that data, so that the data is visible and schema is known_ – We can use a Glue Crawler to create tables in the Glue Catalog
+- _A way to share that data to other AWS accounts_ – AWS Lake Formation grants, as we covered in the previous section
+
+The consuming account will need:
+
+- _A way to query the data in the source account_ – AWS Athena queries on the AWS Lake Formation resource links to the source account.
+
+The diagram below demonstrates how those components can work together for this solution:
+
+<center><img src="/img/lake-formation/01-diagram.png" /></center><br/>
+
+## Setting up Lake Formation
+
+To start with Lake Formation you will need to first assign a Lake Formation Administrator. The administrator will then be able to manage access to resources in the data catalog within the same account and across accounts.
+
+<center><img src="/img/lake-formation/02.png" /></center><br/>
+
+The Lake Formation administrator can either be an IAM user or IAM role. To Setup an IAM role as the administrator, you will need to deploy below CloudFormation code:
+
+```yml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: My data lake source
+
+Resources:
+  LakeformationSettings:
+    Type: AWS::LakeFormation::DataLakeSettings
+    Properties: 
+      Admins: 
+        - DataLakePrincipalIdentifier: arn:aws:iam::100000123000:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_AWSAdministratorAccess_56cabj890003333
+```
+
+To deploy the above CloudFormation stack, store the template in the file `datalake.yml` locally and run the following command:
+
+```sh
+aws cloudformation create-stack --stack-name my-source-glue-stack --template-body file://datalake.yml --capabilities "CAPABILITY_NAMED_IAM"
+```
+
+The second step in setting up lake formation is to change the Lake Formation permission model from IAM to Lake Formation native grants. Unfortunately, this setup is not available via CloudFormation at this time, so we'll need to make the change in the AWS console.
+
+To enable Lake Formation Permissions, go to settings, untick both “Use only IAM access control for new databases” and  “Use only IAM access control for new tables” and press save.
+
+<center><img src="/img/lake-formation/03.png" /></center><br/>
+
+## Establish our source lake
+
+In this blog post we are focusing on the cross-account sharing of Lake Formation, therefore the assumption is you have already ingested your data from the source into an S3 bucket.
+
+Let’s get a starting point, which would be our source bucket and a Glue database for putting our catalog objects into. The following CloudFormation snippet will create those resources:
```

Comment:
"Let's get a starting point.."  maybe change to something like: "Our starting point will be our source bucket and ... "

Reply:
done